doc/rtd/reference/breaking_changes.rst

```diff
@@ -6,6 +6,30 @@ Breaking changes
 This section provides guidance on specific breaking changes to cloud-init
 releases.
 
+.. note::
+    These changes may not be present in all distributions of cloud-init as
+    many operating system vendors patch out breaking changes in
+    cloud-init to ensure consistent behavior on their platform.
+
+Datasource identification
```

Comment:
Grouping these is useful, but it doesn't fit with the existing content since the header `Datasource identification` isn't ordered while the existing content is. Perhaps something like @catmsred's suggestion `23.2-24.1 Datasource identification` would resolve that?